tutorials/sample_vqe_program/qiskit_runtime_vqe_program.ipynb

```diff
@@ -0,0 +1,1394 @@
+   "metadata": {},
+   "source": [
+    "and finally let us specify what comes back when an interm result is returned:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "452dca6f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "meta[\"interim_results\"] = [\n",
+    "    {\"name\": \"params\", \"description\": \"Parameter vector at current optimization step\", \"type\": \"ndarray\"},\n",
+    "  ]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d0328ee9",
+   "metadata": {},
+   "source": [
+    "## Uploading the program\n",
+    "\n",
+    "We now have all the ingredients needed to upload our program.  To do so we need to collect all of our code in one file, here called `sample_vqe.py` for uploading.  This limitation will be removed in later versions of the Runtime.  Alternatively, if the entire code is contained within a single Jupyter notebook cell, then this can be done using the magic function\n",
+    "\n",
+    "```\n",
+    "%%writefile my_program.py\n",
+    "```\n",
+    "\n",
+    "To actually upload the program we need to get a Provider from our IBM Quantum account:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "e65417d4",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<AccountProvider for IBMQ(hub='ibm-q', group='open', project='main')>"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from qiskit import IBMQ\n",
+    "IBMQ.load_account()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "1cacd4d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "provider = IBMQ.get_provider(group='deployed')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "92d17c0f",
+   "metadata": {},
+   "source": [
+    "### Program upload\n",
+    "\n",
+    "The call to `program_upload` takes the target Python file as `data` and the metadata as inputs. **If you have already uploaded the program this will raise an error and you must delete it first to continue**."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "9da62ff0",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'sample-vqe'"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "program_id = provider.runtime.upload_program(data='sample_vqe.py', metadata=meta)\n",
+    "program_id"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ce756bca",
+   "metadata": {},
+   "source": [
+    "Here the returned `program_id` is the same as the program `name` given in the metadata.  However, this need not be the case if there are multiple programs with the same name.  In that case, `program_id` is the unique identifier that needs to be used in calling the program later."
```

Comment:
You can't actually have multiple programs with the same name. It's actually a bit of the opposite. A program name is mapped to a program ID in the format that can be stored in kubernetes. So the same program name is always mapped to the same exact program ID, hence no 2 programs can have the same name. 

On the other side, program names `foo_bar` and `foo-bar` will both be mapped to the program ID `foo-bar`. Hence uploading a program named `foo_bar` will fail if `foo-bar` already exists.

Reply:
Ok so no two names can be the same.  But what happens if someone makes one public with the same name, does that cause trouble?